lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -1092,7 +1092,7 @@ def main():
             type=dict(required=False, default=None, choices=['ethernet', 'team', 'team-slave', 'bond', 'bond-slave', 'bridge', 'vlan'], type='str'),
             ip4=dict(required=False, default=None, type='str'),
             gw4=dict(required=False, default=None, type='str'),
-            dns4=dict(required=False, default=None, type='list'),
+            dns4=dict(required=False, default=[], type='list'),
```

Comment:
There are a lot of `if self.dns4 is not None:` in the code. I wonder if we should change those conditionals too. Or put `if self.dns4 is not None:` for the line that causes the traceback. Thoughts?

Reply:
Yeah, probably even better to continue using `None` but initialize it on [L559](https://github.com/sumkincpp/ansible/blob/5aa09cb5dcd655a60ef787950ca539358b428438/lib/ansible/modules/net_tools/nmcli.py#L559) appropriately.
I'll look onto this.

Reply:
You could probably do something like

```diff
-        self.dns4=' '.join(module.params['dns4'])
+        self.dns4=' '.join(module.params['dns4'] or [])
```

This works fine for me.